File: elixir/two-fer/lib/two_fer.ex
defmodule TwoFer do
  @doc """
  Two-fer or 2-fer is short for two for one. One for you and one for me.
  """


  @spec two_fer(String.t()) :: String.t()
  def two_fer(arg) when not is_binary(arg), do: raise(FunctionClauseError)
  def two_fer(name \\ "") do
      case name == "" do
        true -> "One for you, one for me."
        false -> "One for #{name}, one for me."
      end
  end

end

File: zig/two-fer/two_fer.zig

pub fn twoFer(buffer: []u8, name: ?[]const u8) ![]u8 {
    const name_output = name orelse "you";

    var idx: usize = 0;
    @memcpy(buffer[idx..idx+8], "One for ");
   
    idx += 8;
    @memcpy(buffer[idx..idx+name_output.len], name_output);
    
    idx += name_output.len;
    @memcpy(buffer[idx..idx+13], ", one for me.");

    idx += 13;
    return buffer[0..idx];
}
